src/backend/aspen/api/main.py

```diff
@@ -17,9 +35,23 @@ def get_app() -> FastAPI:
         dependencies=[Depends(partial(set_db, settings))],
     )
 
+    # Configure CORS
+    _app.add_middleware(
+        CORSMiddleware,
+        allow_origins=get_allowed_origins(),
+        allow_headers=["Content-Type"],
+        allow_credentials=True,
+        max_age=600,
+        allow_methods=["*"],
+    )
+    _app.add_middleware(SessionMiddleware, secret_key=settings.SECRET_KEY)
+
     # Warning - do not enable this route!
-    # _app.include_router(users.router, prefix="/v2/users")
+    _app.include_router(
+        users.router, prefix="/v2/users", dependencies=[Depends(get_auth_user)]
```

Comment:
Reading over the FastAPI docs, it seems like they generally [talk about putting the security dependencies into the actual path function](https://fastapi.tiangolo.com/tutorial/security/first-steps/#security-first-steps), rather than handling it at a higher level.

```
@app.get("/items/")
async def read_items(token: str = Depends(oauth2_scheme)):
    return {"token": token}
```

If we want the auth_user aspect to totally global, this is a bit of a hassle. But -- I _think_ -- by doing it this way we can actually use the returned value from the functions we are `Depends`-ing on. So if we have `get_auth_user` return the `found_auth_user` instead, we can effectively make it both param and security declaration on each path function that needs security.

It also looks like you can chain these `Depends` parts way deep. The path functions `Depends` on some callable, but that callable also `Depends` on another thing, etc, etc. So for other paths that rely on auth but are using it to dig up some other set of info than the user id, we could potentially abstract that out. I'm on the fence on whether I like that or not, but it is interesting.

Reply:
Per-path deps are normal for simple/small demo apps, but they've got a doc that talks about more interesting applications, with some examples of adding deps at the routing level:

https://fastapi.tiangolo.com/tutorial/bigger-applications/#include-the-apirouters-for-users-and-items

I agree that being able to use the values directly is nicer though -- I'll look into whether we can actually access dependencies from further up in the call stack.

Reply:
Also, I don't know if we would ever need scopes / permissions, but if we think it's likely we might want to look into the `Security` call instead. It's probably not worth it -- the only real value I can think of in the near-term would be having an `admin` scope -- but if we think it would be useful, it might be nice to learn how to use it and establish the pattern now during the transition.

It took me quite awhile to make sense of how the use the scopes, but I finally made sense of it when I read the code sample under this section: https://fastapi.tiangolo.com/advanced/security/oauth2-scopes/ -- Focus on the parts near the bottom of that giant code-block: `get_current_user`, then `get_current_active_user` and `read_own_items`. I think we wouldn't get our scopes from the authentication process, but when we pull the user's record in the DB, we could effectively get scope info based on `group_admin` and `system_admin`.

Still, I think it's probably not worth it. But if permissioning like that is actually an upcoming need, could be worth the effort (and I'd be happy to work on it as needed).